lang/java/avro/src/test/java/org/apache/avro/TestDataFile.java

```diff
@@ -221,6 +228,37 @@ private void testSyncDiscovery(CodecFactory codec) throws IOException {
         reader.seek(sync);
         assertNotNull(reader.next());
       }
+      // Lastly, confirm that reading (but not decoding) all blocks results in the
+      // same sync points
+      reader.sync(0);
+      ArrayList<Long> syncs2 = new ArrayList<>();
+      while (reader.hasNext()) {
+        syncs2.add(reader.previousSync());
+        reader.nextBlock();
+      }
+      assertEquals(syncs, syncs2);
```

Comment:
This bit tests that reading/skipping blocks yields the same sync points as reading/decoding single records.